content/blog/branch-patterns-automate-prs.md

```diff
@@ -0,0 +1,117 @@
+---
+title: Branch patterns and automating PRs
+date: '2022-04-14T07:00:00.000Z'
+author: James Perkins
+prev: content/blog/read-only-tokens-content-anytime.md
+---
+
+When working on a git backed site you don’t normally want to work directly on your main branch. Ideally, you want to be able to finish your post or site update, check if it looks good, and schedule the release. A good branch pattern and scheduler is key to making this work. In this post, we are going to cover two different options to automate your PRs and an example of the branch pattern I use with Tina.
+
+## Branch Pattern
+
+Branching patterns are very much a personal or business decision, however it usually comes down to how we work on a feature or fix, where we stage the code, and finally when we deploy it. 
+
+My preferred branch pattern looks like the following:
+
+### Permanent Branches:
+
+**main** - This holds the production site that is live and is also the source of truth.
+
+**staging** - This holds the current planned work that is ready for release but may also be a copy of the main branch.
+
+### Temporary Branches:
+
+**feature-name** - Used for a new feature on a site or application.
+
+**bug-fix-name** - Used to fix a bug that was found.
+
+**blog-post-name** - Used for a blog post article. 
+
+When work on a temporary branch is complete, the pull request will go to staging. Here, it will be tested and determined if everything works as expected. A pull request will then be opened to production. Once the production release is complete, the temporary branch will be removed.
```

Comment:
```suggestion
When the work on a temporary branch is complete, the pull request will go to staging. Here, it will be tested and determined if everything works as expected. A pull request will then be opened to production. Once the production release is complete, the temporary branch will be removed.
```